File: data/c#/pattern_4/main.cs
using System;
class Grammind
{
    static void Main()
    {
        int num;
        string input;
        Console.Write("Enter the number of rows and columns for the square: ");
        input = Console.ReadLine();

        while (!int.TryParse(input, out num) || num <= 0)
        {
            Console.Write("Must the number be positive, re-enter the number of rows and columns for the square: ");
            input = Console.ReadLine();
        }

        for (int i = num; i >= 1; i--)
        {
            for (int j = 1; j <= num; j++)
            {
                Console.Write(i + " ");
            }
            Console.Write("\n");
        }
    }
}

File: data/python/pattern_4/main.py
def checkInput(message):
    while True:
        try:
            userInput = int(input(message))
        except ValueError:
            message = 'Must the number be positive, re-enter the number of rows and columns for the square: '
            continue
        else:
            if userInput <= 0:
                message = 'Must the number be positive, re-enter the number of rows and columns for the square: '
                continue
            else:
                return userInput
                break

#MAIN PROGRAM STARTS HERE:
num = checkInput('Enter the number of rows and columns for the square: ')

for x in range(num, 0, -1):
    for y in range(num, 0, -1):
        print ('%d ' % (x), end='')
    print()